src/main/java/com/booking/app/controller/UserController.java

```diff
@@ -0,0 +1,38 @@
+package com.booking.app.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class UserController extends BaseController {
+
+    private static final String API_NAME = "user";
+
+    @GetMapping(UserController.API_NAME + "/{id}")
+    public ResponseEntity<String> getUser(@PathVariable long id) {
+        return ResponseEntity.ok("user1");
+    }
+
+    @GetMapping("/users")
+    public ResponseEntity<String> getAllUsers() {
+        return ResponseEntity.ok("All users!");
+    }
+
+    @PostMapping(UserController.API_NAME)
+    public ResponseEntity<String> createNewUser() {
+        return ResponseEntity.ok("new user");
+    }
+
+    @PutMapping(UserController.API_NAME + "/{id}")
+    public ResponseEntity<String> updateUser(@PathVariable long id){
+        return new ResponseEntity("User " + id + " has been updated!", HttpStatus.ACCEPTED);
```

Comment:
For variables interpolation it is better to use String.format() than classic "+".
For each append that you make, a new String object is created in the String pool

Reply:
Done